File: packages/java/src/main/java/io/github/librandomizer/LibRandomizer.java
package io.github.librandomizer;

import java.security.SecureRandom;

public final class LibRandomizer {
    private static final SecureRandom RNG = new SecureRandom();
    private static final int PRINTABLE_ASCII_MIN = 32;
    private static final int PRINTABLE_ASCII_MAX = 126;

    private LibRandomizer() {
    }

    public static int randomInt() {
        return randomInt(0, 99);
    }

    public static int randomInt(int min, int max) {
        validateRange(min, max);
        return RNG.nextInt((max - min) + 1) + min;
    }

    public static double randomFloat() {
        return randomFloat(0.0, 1.0);
    }

    public static double randomFloat(double min, double max) {
        validateRange(min, max);
        if (min == max) {
            return min;
        }
        return min + RNG.nextDouble() * (max - min);
    }

    public static char randomChar() {
        return randomChar('A', 'Z');
    }

    public static char randomChar(char min, char max) {
        int minCode = validateChar("min", min);
        int maxCode = validateChar("max", max);
        validateRange(minCode, maxCode);
        return (char) randomInt(minCode, maxCode);
    }

    private static void validateRange(double min, double max) {
        if (min > max) {
            throw new IllegalArgumentException("Invalid range: min must be less than or equal to max");
        }
    }

    private static int validateChar(String name, char value) {
        int code = value;
        if (code < PRINTABLE_ASCII_MIN || code > PRINTABLE_ASCII_MAX) {
            throw new IllegalArgumentException("Invalid character range: " + name + " must be printable ASCII");
        }
        return code;
    }
}

File: packages/kotlin/src/main/kotlin/io/github/librandomizer/LibRandomizer.kt
package io.github.librandomizer

import java.security.SecureRandom

object LibRandomizer {
    private val rng = SecureRandom()

    @JvmStatic
    fun randomInt(min: Int = 0, max: Int = 99): Int {
        require(min <= max) { "Invalid range: min must be less than or equal to max" }
        return rng.nextInt((max - min) + 1) + min
    }

    @JvmStatic
    fun randomFloat(min: Double = 0.0, max: Double = 1.0): Double {
        require(min <= max) { "Invalid range: min must be less than or equal to max" }
        if (min == max) return min
        return min + rng.nextDouble() * (max - min)
    }

    @JvmStatic
    fun randomChar(min: Char = 'A', max: Char = 'Z'): Char {
        require(min.code in 32..126) { "Invalid character range: min must be printable ASCII" }
        require(max.code in 32..126) { "Invalid character range: max must be printable ASCII" }
        return randomInt(min.code, max.code).toChar()
    }
}
